### PR TITLE
Hide show xml button from intake flow in demo env; add download pdf button to hub view of state file efile submissions

### PR DESCRIPTION
**Pull request by jenny-heath**

### CODE REVIEW

**Review by mpidcock — APPROVED** [review]
I think it's a little bit odd that we are hiding the pdf also, but that's what the story says. Staging is also "public" - should we hide it there?

**Comment by jenny-heath** [comment]
retrying heroku deploy

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -421,6 +421,11 @@ def acts_like_production?
   end
   helper_method :acts_like_production?
 
+  def show_xml?
+    !(Rails.env.demo? || Rails.env.staging?)
```
Comment (mpidcock) [inline app/controllers/application_controller.rb:425]: Maybe make this an allow list, like only heroku and local. We're now treating it like sensitive data so we should be specific about what envs it should be in. Production should also be excluded anyway.

**Review by mpidcock — CHANGES_REQUESTED** [review]
Replacing `acts_like_production?` with `show_xml?` means the xml button will be introduced to prod, which I don't think we want

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (jenny-heath) [reply app/controllers/application_controller.rb:425]: oh true whitelist is def better

**Pull request merged**